### PR TITLE
Add ios security framework

### PR DESCRIPTION
**Pull request by cammellos**
Fixes: #7226 

Adds sqlcipher to cocoa pods for ios.

It also upgrades status-go with a migration, after this IOS devices might have slower login.

To test:
1) Install previous version on ios (0.9.32)
2) Install same version on a different device and pair the devices.
3) Make sure they are paired.
4) Upgrade ios to this pr
5) Make sure they are still paired ok

Includes kdf changes in status-go can exclude if necessary.
status: ready

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by rasom** [comment]
tested on iphone X: db is encrypted as expected

File: mobile_files/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.9.33
+190109-150228-d0d92a
```
Comment (mandrigin) [inline mobile_files/VERSION]: that looks... weird?

Comment (cammellos) [reply mobile_files/VERSION]: mmh, that indeed looks weird, modified the wrong `VERSION` file :)

Comment (cammellos) [reply mobile_files/VERSION]: fixed

**Comment by annadanchenko** [comment]
@cammellos Android and iOS app crashes if log out second time. It doesn't happen on today's nightly

Steps:
1. Crate account and log in
2. Log out. As a result screen Sign in to Status is shown
3. Log in to Status again
4. Log out -> currently app crashes. Expected: screen Sign in to Status is shown

logs: 
[geth.log](https://github.com/status-im/status-react/files/2759202/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2759203/Status.log)



```
01-15 11:28:18.633 16949 18129 D StatusModule: stopNode
01-15 11:28:18.634 16949 17057 D StatusService: Signal event: {"type":"node.stopped","event":null}
01-15 11:28:18.634 16949 17057 D StatusService: before sendReply true
01-15 11:28:18.634 16949 17069 D StatusModule: Received message: { when=0 what=0 target=im.status.ethereum.module.ServiceConnector$IncomingHandler }
01-15 11:28:18.634 16949 17069 D StatusModule: Signal event: {"type":"node.stopped","event":null}
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: panic: runtime error: invalid memory address or nil pointer dereference
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: [signal SIGSEGV: segmentation violation code=0x1 addr=0
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: x24 pc=0xccf75884]
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: goroutine 3371 [running]:
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/peers.(*Register).Stop(0x0)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: com/status-im/status-go/peers/topic_register.go:46 +0x14
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/node.(*StatusNode).stopDiscovery(0x9873
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 2910, 0x99c70f74, 0x4)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/node/status_node.go:373 +0x20
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: go/node.(*StatusNode).stop(0x98732910, 0xcd814e01, 0x98732910)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/node/status_node.g
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: o:339 +0x1a0
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/node.(*StatusNode).Stop(0x98732910, 0x0, 0x0)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/s
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: tatus-go/node/status_node.go:333 +0x78
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/api.(*StatusBackend).stopNode(0x9851c440, 0x0, 0x0)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: -go/1/src/github.com/status-im/status-go/api/backend.go:194 +0x64
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/api.(*StatusBackend).StopNode(
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 0x9851c440, 0x0, 0x0)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/api/backend.go:186 +0x60
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: github.com/status-im/status-go/api
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: .(*StatusBackend).StopNode-fm(0xcd820508, 0x99a2a7b0)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/lib/library.go:47 +0x1c
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: git
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: hub.com/status-im/status-go/api.RunAsync.func1(0x98b17180, 0x99ad5100)
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/api/utils.
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: go:7 +0x1c
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: created by github.com/status-im/status-go/api.RunAsync
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: 	/ext-go/1/src/github.com/status-im/status-go/api/utils.go:6 
01-15 11:28:18.638 16949 17047 D StatusNativeLogs: +0x54
```

**Comment by cammellos** [comment]
thanks @annadanchenko I will take a look, in the meantime cc @dshulyak

**Comment by annadanchenko** [comment]
UPDATED info: same issue exists for nightly build + beta or staging fleet https://github.com/status-im/status-react/issues/7282 so not a bug introduced here

Initial info:
@cammellos I'm testing this PR using staging fleet with same status-go there on real Android 6.0.1, Galaxy S6
Sometimes when join new public chat I see no historical messages (even if request 24 hour history) and geth.log contains lots of
``` t=2019-01-15T15:09:48+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="unexpected EOF"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="unexpected EOF"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="unexpected EOF"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="unexpected EOF"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="proto: can't skip unknown wire type 4"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="unexpected EOF"
t=2019-01-15T15:09:49+0000 lvl=eror msg="Failed handling message with error" package=status-go/services/sshext.PublicAPI err="proto: can't skip unknown wire type 6"
```
[geth.log](https://github.com/status-im/status-react/files/2760241/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2760242/Status.log)

I can't reproduce it constantly but if try these steps 3-4 times you might be lucky:
1. create new account
2. add picture to profile (to allow access to Storage)
3. change fleet to staging
4. log in
5. join "status" or "status-core" or "testanna" chat and check if history is retrieved

I don't think it's because of mailserver. When no messages were retrieved there was "mail-01.do-ams3.eth.staging" mailserver. I've killed and started the app again, other mailserver was selected and history was retrieved. Then I've manually selected  "mail-01.do-ams3.eth.staging" and joined another public chat - history is retrieved fine.

Will try to reproduce same on nightly build (to see if it's related to status-go update)

**Comment by asemiankevich** [comment]
i dont see offline inbox messages in the following case:

- 2 devices in contacts, iOS and android
- ios goes offline
- android sends some messages to ios
- ios gets back online and taps the push notification to be redirected to the messages sent offline

As a result i see PN coming to iOS but no messages that were sent while iOS was offline. In order to receive them i have to restart the app.

fleet: staging
os: ios 11.4.1, android 8.

**Comment by Serhy** [comment]
Above scenario, https://github.com/status-im/status-react/pull/7229#issuecomment-454843358 looks OK for me in most cases. I've seen it twice (from ~10 time going offline-online) that after tap on PN and navigating to 1-1 chat no offline messages retrieved but after investigation realised that in public chats there was the same issue and I realised that was what https://github.com/status-im/status-react/pull/7229#issuecomment-454344203 + https://github.com/status-im/status-react/issues/7282 

So after re-login to account (relogin to the same account or login to another acc) there will be no offline inboxed messages in chat. And the second log out action crashes the app. 

However, all above is fine with fix @cammellos made here https://github.com/status-im/status-react/pull/7285 

Just thought, @asemiankevich what happens if you try to log out after navigating to chat and seen no messages from offline history? If it crashes the app then it may be issue that https://github.com/status-im/status-react/pull/7285 is about to fix.

**Comment by asemiankevich** [comment]
> Above scenario, [#7229 (comment)](https://github.com/status-im/status-react/pull/7229#issuecomment-454843358) looks OK for me in most cases. I've seen it twice (from ~10 time going offline-online) that after tap on PN and navigating to 1-1 chat no offline messages retrieved but after investigation realised that in public chats there was the same issue and I realised that was what [#7229 (comment)](https://github.com/status-im/status-react/pull/7229#issuecomment-454344203) + #7282
> 
> So after re-login to account (relogin to the same account or login to another acc) there will be no offline inboxed messages in chat. And the second log out action crashes the app.
> 
> However, all above is fine with fix @cammellos made here #7285
> 
> Just thought, @asemiankevich what happens if you try to log out after navigating to chat and seen no messages from offline history? If it crashes the app then it may be issue that #7285 is about to fix.

@Serhy second logout in a row crashes the app no matter of the messages

**Comment by asemiankevich** [comment]
I have reproduced this case with

- current PR build , new status-go and staging fleet
- current PR build , new status-go and beta fleet

i also have checked these combinations:
- nightly + staging fleet
- nightly + beta fleet

When testing with nightlies i see this: after i tap the PN on device - i am redirected to the chat and i see 'Offline' tooltip and no offline inbox messages. However it disappears in a few seconds, and then offline messages are fetched.

to summarize this - i feel like this PR + staging fleet is worse in comparison with nightly build.

What exactly i did:

1. Added 2 devices to contacts
2. Changed fleet to staging
3. Turned off wifi on android
4. Sent 10 messages (from 1 to 10) from ios to android
5. Turned on wifi on android
6. Wait a bit while android gets Push Notifications for the messages that were sent offline
7. Tap the push notification on android

As a result i see only messages `9` and `10` in the chat. No `fetching messages`appears.

NOTE: messages are fetched after i logged out and relogin.

[geth.log](https://github.com/status-im/status-react/files/2768039/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2768040/Status.log)

[geth.log](https://github.com/status-im/status-react/files/2768081/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2768082/Status.log)

**Comment by annadanchenko** [comment]
@cammellos pairing is not working after upgrade to this PR
Steps:
1. Install previous version on ios (0.9.32)
2. Install same version on a different device and pair the devices (used Android device to pair)
3. Make sure they are paired. I've added another contact from iOS device and sent message in 1:1 from iOS, on paired device new 1:1 chat is shown with sent message
3. Upgrade ios to this pr
4. Make sure they are still paired ok. In my case if iOS sends new message in 1:1 chat then on paired device it's not shown (as it was before upgrade)

checked upgrades to build 10 and to build 11

geth.log from iOS 
[geth.log](https://github.com/status-im/status-react/files/2769896/geth.log)

**Comment by adambabik** [comment]
The problem with crashes is fixed with 0.19.0-beta.6?

**Comment by mandrigin** [comment]
@adambabik it works for me... but one thing though is that we need to revert a revert of Pedro's changes because the PN update was finally merged :)

**Comment by adambabik** [comment]
@mandrigin oh man :D We will never merge this. If that's necessary, let's do it. Is it https://github.com/status-im/status-go/pull/1352?

**Comment by mandrigin** [comment]
@adambabik heh, it is nothing compared to this PN refactoring effort we merged :-D
but on the next time we shouldn't even allow to break API compatibility if that is avoidable.

**Comment by mandrigin** [comment]
@adambabik I think it is this commit we need to revert in status-go: https://github.com/status-im/status-go/commit/61bbbdde58bba0f6f24e51b6580ba8be96765db5

**Comment by mandrigin** [comment]
according to what I see in the native code of the status-react, it is that commit to revert indeed

**Comment by adambabik** [comment]
@mandrigin oh ok so we need to revert the revert but how is https://github.com/status-im/status-go/pull/1352 related to that? Or it's something totally separate?

**Comment by mandrigin** [comment]
@adambabik it brings back the reverted finctionality but by adding a new method instead of changing the existing one. It should be pointless right now.

**Comment by annadanchenko** [comment]
> The problem with crashes is fixed with 0.19.0-beta.6?

yes, https://github.com/status-im/status-react/pull/7229#issuecomment-454344203 is  fixed (checked on Android and iOS build 11)

**Comment by cammellos** [comment]
closing as already in https://github.com/status-im/status-react/pull/7294

**Pull request closed without merging**